subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/ivy/IvyCustomStatusLatestVersionIntegrationTest.groovy

```diff
@@ -70,14 +75,28 @@ repositories {
     }
 }
 configurations { compile }
+
+class StatusRule implements ComponentMetadataRule {
+
+    String releaseVersion
+    
+    public StatusRule(String releaseVersion) {
+        this.releaseVersion = releaseVersion
+    }
+
+    public void execute(ComponentMetadataContext context) {
+            if (context.details.id.version == releaseVersion) {
+                context.details.status = 'release'
+            }
+    }
+}
+
 dependencies {
     compile 'org.test:projectA:latest.release'
     components {
-        all { ComponentMetadataDetails details ->
-            if (details.id.version == project.properties['releaseVersion']) {
-                details.status = 'release'
-            }
-        }
+        all(StatusRule, {
+            params(project.properties['releaseVersion'] == null ? '' : project.properties['releaseVersion'])
```

Comment:
Looks like having a null as a `params` argument causes an NPE, probably linked to the vararg there.